File: frontend/src/types/research.ts
import type { ISODateString, Verdict } from "./common";

export interface AgentMessage {
  id: string;
  agent: "bull" | "bear";
  company: string;
  ticker: string;
  text: string;
  timestamp: ISODateString;
  citations: string[];
}

export type AgentStatus = "waiting" | "active" | "concluded";

export interface AgentState {
  status: AgentStatus;
  currentCompany?: string;
  currentTicker?: string;
  currentHypothesis?: string;
  evidenceCount: number;
  messages: AgentMessage[];
}

export interface ManagerEvaluation {
  thesisStrength: number;
  evidenceQuality: number;
  upsidePotential: number;
  risk: number;
  verdict: Verdict;
  reasons: string[];
  invalidationConditions: string[];
}

export interface ResearchResult {
  id: string;
  company: string;
  ticker: string;
  sector: string;
  price: number;
  marketCapCr: number;
  event: string;
  eventDate: ISODateString;
  significance: string;
  bullScore: number;
  bearScore: number;
  managerScore: number;
  potentialReturnPct: number;
  riskScore: number;
  confidencePct: number;
  evidenceCount: number;
  verdict: Verdict;
  manager: ManagerEvaluation;
  bullThesis: string[];
  bearThesis: string[];
  evidence: EvidenceItem[];
  historicalComparables: HistoricalComparable[];
  news: NewsItem[];
}

export interface EvidenceItem {
  id: string;
  label: string;
  source: string;
  strength: "supporting" | "contradicting" | "neutral";
  citedBy: "bull" | "bear" | "verification";
}

export interface HistoricalComparable {
  company: string;
  event: string;
  outcome: string;
  returnPct: number;
  period: string;
}

export interface NewsItem {
  id: string;
  headline: string;
  source: string;
  publishedAt: ISODateString;
  summary: string;
}

export interface ResearchRunSummary {
  articlesScanned: number;
  eventsDetected: number;
  candidatesSelected: number;
  runDurationSec: number;
}

File: backend/app/schemas/research.py
from pydantic import BaseModel, ConfigDict, Field

Verdict = str  # "BUY" | "WATCH" | "PASS"


class ManagerEvaluation(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    thesis_strength: float = Field(alias="thesisStrength")
    evidence_quality: float = Field(alias="evidenceQuality")
    upside_potential: float = Field(alias="upsidePotential")
    risk: float
    verdict: Verdict
    reasons: list[str]
    invalidation_conditions: list[str] = Field(alias="invalidationConditions")


class EvidenceItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    id: str
    label: str
    source: str
    strength: str
    cited_by: str = Field(alias="citedBy")


class HistoricalComparable(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    company: str
    event: str
    outcome: str
    return_pct: float = Field(alias="returnPct")
    period: str


class NewsItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    id: str
    headline: str
    source: str
    published_at: str = Field(alias="publishedAt")
    summary: str


class ResearchResult(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    id: str
    company: str
    ticker: str
    sector: str
    price: float
    market_cap_cr: float = Field(alias="marketCapCr")
    event: str
    event_date: str = Field(alias="eventDate")
    significance: str
    bull_score: float = Field(alias="bullScore")
    bear_score: float = Field(alias="bearScore")
    manager_score: float = Field(alias="managerScore")
    potential_return_pct: float = Field(alias="potentialReturnPct")
    risk_score: float = Field(alias="riskScore")
    confidence_pct: float = Field(alias="confidencePct")
    evidence_count: int = Field(alias="evidenceCount")
    verdict: Verdict
    manager: ManagerEvaluation
    bull_thesis: list[str] = Field(alias="bullThesis")
    bear_thesis: list[str] = Field(alias="bearThesis")
    evidence: list[EvidenceItem]
    historical_comparables: list[HistoricalComparable] = Field(alias="historicalComparables")
    news: list[NewsItem]


class ResearchRunSummary(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    articles_scanned: int = Field(alias="articlesScanned")
    events_detected: int = Field(alias="eventsDetected")
    candidates_selected: int = Field(alias="candidatesSelected")
    run_duration_sec: float = Field(alias="runDurationSec")


class StartRunResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    run_id: str = Field(alias="runId")
